docs/canonicalk8s/capi/howto/rollout-upgrades.md

```diff
@@ -80,6 +80,16 @@ kubectl --kubeconfig c1-kubeconfig.yaml get nodes -o wide
 The machines will be replaced in turn until all machines run on
 the desired version.
 
+## Configure the machine count during the control plane upgrade
```

Comment:
Am I right in assuming that this spec.strategy.rollingUpdate.maxSurge must be set in the config before the upgrade is triggered? If so I think this section could be moved up to above ## Monitor the control plane upgrade

Reply:
Considering that the default value is the backwards compatible one, spec.strategy.rollingUpdate.maxSurge = 1, the user that deploys the cluster via cluster api has to plan ahead before the deployment in order to use the spec.strategy.rollingUpdate.maxSurge = 0. This is an actual good question - I suppose you refer to the situation where spec.strategy.rollingUpdate.maxSurge is not set (defaulted to 1) or set to value 1 (same scenario) -> the cluster is deployed -> and then spec.strategy.rollingUpdate.maxSurge is set to 0. What happens in that scenario? Will a rolling upgrade be triggered? If one other spec config is changed as the same time, the rolling upgrade will be triggered and will it be done with the new workflow (no extra spare)? I need to confirm this locally and will get back.

Reply:
Verified and confirmed the following:

      * what happens if maxSurge is set to 0, and then maxSurge is set to 1?
        1. will it trigger an rolling upgrade?
          DONE -> NO, it does not trigger a rolling upgrade when changing the maxSurge values
        2. if not, given the maxSurge is 1 and then is set to 0 and the config has another update that will actually trigger a rolling upgrade?
          * if so, will it trigger a rolling upgrade using 0?
          DONE= -> YES, if both configs (the one for maxSurge=0, and for example, kubelet_extra_args are changed), it will triggere a rolling upgrade using maxSurge=0

Reply:
> Am I right in assuming that this spec.strategy.rollingUpdate.maxSurge must be set in the config before the upgrade is triggered? If so I think this section could be moved up to above ## Monitor the control plane upgrade

The spec.strategy.rollingUpdate.maxSurge can be set in the config when the upgrade is triggered, as long as at the same time, there is a trigger for a rollingUpgrade, like changing another spec config like the kubelet_extra_args.